marklogic-rdf4j/src/main/java/com/marklogic/semantics/rdf4j/MarkLogicRepositoryConnection.java

```diff
@@ -676,7 +677,7 @@ public boolean hasStatement(Resource subject, IRI predicate, Value object, boole
             logger.debug(queryString);
             MarkLogicBooleanQuery query = prepareBooleanQuery(queryString); // baseuri ?
 
-            setBindings(query, subject, predicate, object, contexts);
+            setBindings(query, subject, predicate, Util.getInstance().skolemize(object), contexts);
```

Comment:
That's not what you want though.  The skolemization would have to be 100% deterministic to serve the needs here.  A bnode is required to be unique within a single transaction, but always generated anew with subsequent ones.  So you cannot persist them without turning them into real IRIs, hence skolemization.  But that means if you ingest the same graph over and over again, with bnodes, then certain relationships will get duplicated again and again.  It's a classic RDF challenge.